pkg/prometheus/resource_selector.go

```diff
@@ -1327,6 +1327,11 @@ func (rs *ResourceSelector) validateLinodeSDConfigs(ctx context.Context, sc *mon
 	return nil
 }
 func (rs *ResourceSelector) validateKumaSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+
+	if rs.version.LT(semver.MustParse("2.50.0")) {
```

Comment:
Maybe KumaSD is added to older versions
ref: https://github.com/prometheus/prometheus/blame/350d591e3b1008c406e1ff880230c498ed4189a5/discovery/xds/kuma.go